datafusion-postgres/src/handlers.rs

```diff
@@ -43,9 +43,16 @@ pub struct HandlerFactory {
 }
 
 impl HandlerFactory {
-    pub fn new(session_context: Arc<SessionContext>, auth_manager: Arc<AuthManager>) -> Self {
-        let session_service =
-            Arc::new(DfSessionService::new(session_context, auth_manager.clone()));
+    pub fn new(
+        session_context: Arc<SessionContext>,
+        auth_manager: Arc<AuthManager>,
+        query_timeout: Option<std::time::Duration>,
```

Comment:
I suggest you to put this into a new pull request. Also it should be configured at per-session level using `SET statement_timeout`